packages/puppeteer-core/src/bidi/Connection.ts

```diff
@@ -24,21 +26,68 @@ import type {
 const debugProtocolSend = debug('puppeteer:webDriverBiDi:SEND ►');
 const debugProtocolReceive = debug('puppeteer:webDriverBiDi:RECV ◀');
 
+// Start of CDP specific non-standard protocol extensions.
```

Comment:
Why do we need to build this in Puppeteer rather then rely on the Chromium-BiDi types?